Kitodo/src/main/java/org/kitodo/production/KitodoProduction.java

```diff
@@ -0,0 +1,157 @@
+import org.kitodo.production.security.SecurityUserDetails;
+import org.kitodo.production.services.ServiceManager;
+import org.springframework.security.core.context.SecurityContextImpl;
+
+/**
+ * Kitodo.Production is the workflow management module of the Kitodo suite. It
+ * supports the digitization process of various types of materials.
+ */
+@WebListener
+public class KitodoProduction implements ServletContextListener, HttpSessionListener {
+    private static final Logger logger = LogManager.getLogger(KitodoProduction.class);
+    private static CompletableFuture<KitodoProduction> instance = new CompletableFuture<>();
+    private ServletContext context;
+    private Optional<Manifest> manifest;
+    private KitodoVersion version = new KitodoVersion();
+    private ActiveMQDirector activeMQDirector;
+
+    /* This is the main() function. The class is instantiated by the servlet
+     * container, which then calls this function. */
+    @Override
+    public void contextInitialized(ServletContextEvent event) {
+        context = event.getServletContext();
+        manifest = retrieveManifestFileAsStream(context);
+        manifest.ifPresent(version::setupFromManifest);
+        instance.complete(this);
+        startActiveMQ();
+    }
+
+    private static final Optional<Manifest> retrieveManifestFileAsStream(ServletContext context) {
+        try (InputStream manifestResource = context.getResourceAsStream("/META-INF/MANIFEST.MF")) {
+            if (Objects.nonNull(manifestResource)) {
+                return Optional.of(new Manifest(manifestResource));
+            }
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+            context.log(e.getMessage());
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Start up the active MQ connection.
+     */
+    private void startActiveMQ() {
+        if (ConfigCore.getOptionalString(ParameterCore.ACTIVE_MQ_HOST_URL).isPresent()) {
+            activeMQDirector = new ActiveMQDirector();
+            Thread connectAsynchronously = new Thread(activeMQDirector);
+            connectAsynchronously.setName(ActiveMQDirector.class.getSimpleName());
+            connectAsynchronously.setDaemon(true);
+            connectAsynchronously.start();
+        }
+    }
+
+    /**
+     * Returns the application’s main class.
+     * 
+     * @return the main class
+     * @throws UndeclaredThrowableException
+     *             if the servlet container is shut down before the web
+     *             application is started
+     */
+    public static KitodoProduction getInstance() {
+        try {
+            return instance.get();
+        } catch (InterruptedException | ExecutionException e) {
+            logger.fatal(e);
+            throw new UndeclaredThrowableException(e);
+        }
+    }
+
+    /**
+     * Returns the servlet context.
+     * 
+     * @return the servlet context
+     */
+    public ServletContext getServletContext() {
```

Comment:
Do we need this method? My IDE shows it is unused, and the two interfaces that this method implements do not seem to require the method either.

Reply:
It's just a getter. You might want to get the servlet context to get the application's working directory, for example, or the server version. It's here for convenience. At the moment, it's not used yet.

Reply:
Please remove the method if it's unused and not required at the moment. Doesn't matter if it's just a getter. It should be added if required in the future.